feature_engine/datetime/datetime.py

```diff
@@ -343,19 +344,17 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
 
         return X
 
+    @Substitution(
+        input_features=_input_features_docstring,
+    )
     def get_feature_names_out(
         self, input_features: Optional[Union[List, str]] = None
     ) -> List:
         """Get output feature names for transformation.
 
         Parameters
         ----------
-        input_features: str, list, default=None
-            Input features. If `None`, then the names of all the variables in the
-            transformed dataset (original + new variables) is returned. If 'index',
-            then the name of the features created from the index is returned. If list
-            with feature names, the names for the datetime features derived from
-            the variables in the list will be returned."
+        {input_features}
```

Comment:
I would not substitute this method for a common string, because datetime features has its own particular functionality. I would leave the original docstring.